jmetal-core/src/main/java/org/uma/jmetal/measure/impl/MeasureFactory.java

```diff
@@ -71,11 +71,25 @@ public Value get() {
 				return (Value) cache[0];
 			}
 
-			@Override
+			
+			/*
+			 * 
+			 * (non-Javadoc)
+			 * The Object.finalize() method is called on an object by the garbage 
+			 * collector when it determines that there are no more references to 
+			 * the object. But there is absolutely no warranty that this method 
+			 * will be called AS SOON AS the last references to the object are 
+			 * removed. It can be few microseconds to few minutes later. So when 
+			 * system resources need to be disposed by an object, it's better to 
+			 * not rely on this asynchronous mechanism to dispose them.
+			 */
+			
+		/*	@Override
 			protected void finalize() throws Throwable {
 				push.unregister(listener);
 				super.finalize();
 			}
+		*/
```

Comment:
This `finalize` is there because of the implementation choices made: the pull measure is based on a push by registering to it. It should be unregistered at some points and, given the actual design, the best place is in a finalizer. With this change, we guarantee that we never unregister, and thus always maintain the pull measure in memory as long as the push measure exists. This can have a significant impact on performance.

`finalize` should be avoided when we can do better, but not removed because "it should not be used".